Automattic/wp-calypso
# Dev Tools: Load `<ReauthRequired />` for Account Settings Helper when in view

From p1683880137480689-slack-C02AED43D

## Proposed Changes

Loads the `<ReauthRequired />` component when the Account Settings Helper is in view:

<img width="462" alt="image" src="https://github.com/Automattic/wp-calypso/assets/36432/bee213b1-eef2-4a97-90c7-3635d6e58e6c">

The 'Language Picker' requires a valid two step cookie.

## Testing Instructions

1. Delete the `twostep_auth` cookie for `public-api.wordpress.com`.
2. Navigate to `/me` and verify the two step reauth modal appears (but don't complete it.
3. Navigate to `/sites` and verify the two step reauth modal doesn't appear.
4. Hover over 'Account Settings' in Dev Tools.
5. Verify the two step reauth modal appears.
6. Complete the two step reauth modal.
7. Refresh the page and verify the modal no longer appears when you hover over 'Account Settings' in Dev Tools.
8. Navigate back to `/me` and verify the reauth modal no longer appears.
